### DISCUSSION TITLE
Flashing the Crazyflie Bolt

### DISCUSSION THREAD

**Discussion by tedzaremba** [bitcraze/discussions#1241] [Q&A] [answered]
Is it possible to flash custom firmware on the Crazflie Bolt through a wired connection? If not, I was going to order this radio to flash wirelessly (https://store.bitcraze.io/products/crazyradio-2-0).

**Comment by tobbeanton** [comment, marked as answer]
It is possible to flash the STM32F405 using [dfu](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/development/dfu/) which contains the firmware. To update the radio MCU, nrf51822, the crazyradio or debugger is needed.

**Reply by tedzaremba** [reply]
Thanks for the tip. I'm using the bitcraze vm to flash it, Is this what I should see when it's finished flashing?
![image](https://github.com/bitcraze/discussions/assets/114715499/d7d7e785-4dd0-4e77-8fb9-1d9e1d34d087)

**Reply by tedzaremba** [reply]
Never mind, there's a bug where it doesn't wait for it to connect back to the vm once it's in bootloader mode. Got around this by connecting it to my vm and running 'make flash_dfu_manual'
